lib/msg-format.c

```diff
@@ -94,7 +116,7 @@ msg_format_parse(MsgFormatOptions *options, const guchar *data, gsize length, Lo
 
   msg_trace("Initial message parsing follows");
   msg_format_preprocess_message(options, data, length, msg);
-  options->format_handler->parse(options, data, length, msg);
+  msg_format_process_message(options, data, length, msg);
```

Comment:
In case of `no-parse` flag, we still call `msg_format_postprocess_message`, and thus setting additional flags from config (e.g. assume-utf8, local).
I'm okay with this, but previously neither `syslog-format`, nor `linux-kmsg` set these flags.

Reply:
The behavioral change was intentional, and again it is the "should have been this way" case.

The utf8 validation/sanitization is not yet centralized (but probably should be), I didn't touch that because of the performance implications.

I could easily centralize assume-utf8, so I did that. And the message can be utf-8 even if it wasn't parsed.

The local flag was set even previously depending on the LP_LOCAL flag for the source, which is done by the file() source.